### PR TITLE
Decoupling mongo db dependencies to allow adding postgreSQL

### PR DESCRIPTION
**Pull request by DavidGershony**

### CODE REVIEW

File: src/Blockcore.Indexer.Core/Operations/IUtxoCache.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-using System.Collections.Generic;
-using Blockcore.Indexer.Core.Operations.Types;
-using Blockcore.Indexer.Core.Storage.Mongo.Types;
+// using System.Collections.Generic;
+// using Blockcore.Indexer.Core.Operations.Types;
```
Comment (dangershony) [inline src/Blockcore.Indexer.Core/Operations/IUtxoCache.cs:2]: why not delete this?

File: src/Blockcore.Indexer.Core/Operations/Types/InsertStats.cs [hunk 2]
```diff
@@ -23,10 +20,5 @@ public class InsertStats
       /// Gets or sets the outputs.
       /// </summary>
       public int InputsOutputs { get; set; }
-
-      /// <summary>
-      /// Gets or sets the items.
-      /// </summary>
-      public List<MempoolTable> Items { get; set; }
```
Comment (dangershony) [inline src/Blockcore.Indexer.Core/Operations/Types/InsertStats.cs:30]: is this not used?

File: src/Blockcore.Indexer.Core/Operations/Types/UtxoCache.cs [hunk 3]
```diff
@@ -14,44 +14,44 @@ public class UtxoCacheItem
...
-      }
-
-      public void RemoveFromCache(IEnumerable<InputTable> inputs)
-      {
-         foreach (InputTable output in inputs)
-         {
-            cache.TryRemove($"{output.Outpoint.TransactionId}-{output.Outpoint.OutputIndex}", out _);
-         }
-      }
+      // private readonly ILogger<UtxoCache> logger;
```
Comment (dangershony) [inline src/Blockcore.Indexer.Core/Operations/Types/UtxoCache.cs:17]: delete?

**Review by dangershony — COMMENTED** [review]
This PR looks ok to me, did you manage to test starting it from zero? (building the indexes)

File: src/Blockcore.Indexer.Core/Operations/Types/InsertStats.cs (hunk 2 above)
Comment (DavidGershony) [reply src/Blockcore.Indexer.Core/Operations/Types/InsertStats.cs:30]: Done

File: src/Blockcore.Indexer.Core/Operations/Types/UtxoCache.cs (hunk 3 above)
Comment (DavidGershony) [reply src/Blockcore.Indexer.Core/Operations/Types/UtxoCache.cs]: Done

**Review by dangershony — APPROVED** [review]

**Pull request merged**